### DISCUSSION TITLE
UX feedback report

### DISCUSSION THREAD

**Discussion by ggdorman** [Tunous/Dawn#525] [Ideas]
Hi,
Instead of opening a slew of new issues, I'd like to give my feedback on the user experience on the app in the hopes of improving the app. I've also referenced the existing issue reports that I could find.
Here are the main issues I found while using the app:
- Reddit Video media plays back in low-res (#265)
- Reddit Video media downloads have no sound (#309)
- When returning to a subreddit feed after opening a post, the post becomes temporarily invisible
- A button should be implemented in the subreddit edit list to quickly remove all the default subs (it is very cumbersome to delete them all by hand)
- In a subreddit feed, there is no visible distinction between media types- videos, gifs, photos are all displayed the same
- Downloaded media creates a 0 byte ghost copy in /Internal Memory/Pictures/
- It would be nice to have comment navigation buttons (up and down arrows for navigating to the next and previous comment thread)
- It would be nice to show multiple subreddits combined in one feed (see https://github.com/Tunous/Dawn/issues/249#issuecomment-643806276)
- The app should also register as being able to open libreddit links (#508)

Solving these problems would make the app significantly more useful and greatly improve user experience. I hope you will consider my suggestions, and thanks for maintaining this wonderful app :).

Cheers!

**Comment by AbsurdlySuspicious** [comment]
> Reddit Video media plays back in low-res #265 

The cause for this is still not clear, if you could provide us some link(s) to posts where difference is clearly distinguishable that would be very useful

 > When returning to a subreddit feed after opening a post, the post becomes temporarily invisible

I believe this happens only when previous activity is destroyed and unloaded from memory, so it has to restore itself and load content again. There isn't much we can do about this aside from putting loading indicator here. I don't think putting whole post list to saved instance state is a good idea  
EDIT: maybe we can put sub post list to cache in similiar fashion as it works in main feed and keep it for some time

> remove all the default subs

I'm not sure I understand what are you referring to. You mean unsubscribe from all subs?

> Downloaded media creates a 0 byte ghost copy 

You mean every time or only after failed downloads?

**Reply by ggdorman** [reply]
> The cause for this is still not clear, if you could provide us some link(s) to posts where difference is clearly distinguishable that would be very useful

The low-res issue affects every single video, an example would be [redd.it/qstkja](https://redd.it/qstkja/).

> EDIT: maybe we can put sub post list to cache in similiar fashion as it works in main feed and keep it for some time

Sounds good 👍

> I'm not sure I understand what are you referring to. You mean unsubscribe from all subs?

The app has a long list of default subreddits in the subreddit navigation area (I don't know what to call it, the one with frontpage, all, etc.). I only want to have a few subs in there that I actively browse, so it would be really usefull to have an option to delete all the subs from the list (one implementation could be: select all button --> delete button).

> You mean every time or only after failed downloads?

Every time. I just double-checked and the file itself doesn't appear to exist, though it shows up in all gallery apps I tested. The exact path is /storage/emulated/0/Pictures/<picture>.jpg. This is the only app I've seen with this problem so it may have something to do with the way it saves media/interfaces with the MediaStore API.

**Comment by AbsurdlySuspicious** [comment]
(to devs) Things to note that don't have issues yet:
- It would be nice to have comment navigation buttons (up and down arrows for navigating to the next and previous comment thread)
- ~~In a subreddit feed, there is no visible distinction between media types- videos, gifs, photos are all displayed the same~~ #529

**Reply by ggdorman** [reply]
I came up with a quick little solution for distinguishing media types: you could add a little badge or text to the post titlebar, on the same row where it shows the comment count and sub, anchored to the right. This way it is still visible, but out of the way, keeping the interface minimal and not covering the content.
Example badges would be: [IMAGE] [VIDEO] [EXTERNAL]

**Comment by AbsurdlySuspicious** [comment]
New issues: #526 #527 #528 #529
